Sources/Rendering/OpenGL/VolumeMapper/index.js

```diff
@@ -904,7 +913,20 @@ function vtkOpenGLVolumeMapper(publicAPI, model) {
       actor.getProperty().getShade() ||
       actor.getProperty().getUseGradientOpacity(0)
     ) {
-      model.lightingTexture.activate();
+      // Only activate once volumeInfo has been populated.
+      if (model.lightingTexture.getComputedGradients()) {
+        model.lightingTexture.activate();
+      } else {
+        // We wanted to render, but the gradients have not finished computing.
+        // So, re-render later.
+        if (computedGradientsRenderTimeout !== null) {
+          clearTimeout(computedGradientsRenderTimeout);
+        }
+        computedGradientsRenderTimeout = setTimeout(
+          model.openGLRenderWindow.modified,
+          20
+        );
```

Comment:
I'm not sure I get what this is doing in the timeout. Marking the renderWindow modified shouldn't change anything last I knew.

Reply:
I should make a comment here, but this triggers a render after the gradients are ready.